### PR TITLE
fix(regex): don’t escape forward slash in fileMatch

### LINKED ISSUE #19312: Inconsistent escaping in fileMatch
### Describe the proposed change(s).

`fileMatch: ['(^|\\/)`:

image

There are then 49 instances of `fileMatch: ['(^|/)`.

Do both work?

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Removes unnecessary forward slash escaping.

## Context

Closes #19312

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/flux/common.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { regEx } from '../../../util/regex';
 
 export const systemManifestRegex =
-  '(^|\\/)flux-system\\/(?:.+\\/)?gotk-components\\.yaml$';
+  '(^|/)flux-system\\/(?:.+\\/)?gotk-components\\.yaml$';
```
Comment (viceice) [inline lib/modules/manager/flux/common.ts:4]:
```suggestion
  '(^|/)flux-system/(?:.+/)?gotk-components\\.yaml$';
```
should remove all escapes

**Review by viceice — COMMENTED** [review]
we should search for additional `\\/` to replace

**Comment by Churro** [comment]
This branch doesn't yet see two more patterns that PR #19182 introduced in [lib/modules/manager/gradle/index.ts](https://github.com/renovatebot/renovate/commit/0d9aaed601049fd93dcf51fdb8bf14c3e5078840#diff-8d1fda1d67471e03db8ff8dac78c4c2556866ab158ffc066686015abf2433f27) just today. Probably also worth considering in this PR👍

**Comment by rarkins** [comment]
Now replaced all, but I'd like to be paranoid and have multiple eyes on this.

File: lib/config/options/index.ts [hunk 2]
```diff
@@ -944,7 +944,7 @@ const options: RenovateOptions[] = [
   {
     name: 'matchBaseBranches',
     description:
-      'List of strings containing exact matches (e.g. `["main"]`) and/or regex expressions (e.g. `["/^release\\/.*/"]`). Valid only within a `packageRules` object.',
+      'List of strings containing exact matches (e.g. `["main"]`) and/or regex expressions (e.g. `["/^release/.*/"]`). Valid only within a `packageRules` object.',
```
Comment (viceice) [inline lib/config/options/index.ts:947]: Do we trim the leading and trailing slash before passing to regex function? those can maybe change behavior. 🤔

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/config/options/index.ts (hunk 2 above)
Comment (rarkins) [reply lib/config/options/index.ts:947]: Yes, we do

**Review by viceice — DISMISSED** [review]

**Comment by rarkins** [comment]
@JamieMagee another set of eyes please :)

**Review by JamieMagee — DISMISSED** [review]
This looks okay, but we might need to document the trimming of `/` in regex patterns. The standard[^1] says that the first character of a regex is the delimiter, but the delimiter can be any character. `/` is just a convention, but I've seen `#`, `@`, etc. used, which makes it easier when `/` is a part of the pattern you're trying to match.

[^1]: https://en.wikipedia.org/wiki/Regular_expression#Delimiters

**Comment by rarkins** [comment]
I think we've mentioned that `/ ... /` or `!/ ... /` are our conventions whenever a field supports it, but we can double check

**Comment by rarkins** [comment]
Non-breaking but will release it as part of v35 just in case

File: lib/modules/manager/woodpecker/index.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ export const language: ProgrammingLanguage = 'docker';
 export { extractPackageFile };
 
 export const defaultConfig = {
-  fileMatch: ['^\\.woodpecker(?:\\/[^/]+)?\\.ya?ml$'],
+  fileMatch: ['(^|/)\\.woodpecker(?:/[^/]+)?\\.ya?ml$'],
```
Comment (viceice) [inline lib/modules/manager/woodpecker/index.ts:10]:
```suggestion
  fileMatch: ['^\\.woodpecker(?:/[^/]+)?\\.ya?ml$'],
```
only supported on root folder from woodpecker

**Review by viceice — APPROVED** [review]

**Review by JamieMagee — APPROVED** [review]

**Comment by JamieMagee** [comment]
There are some docs updates, but they won't block code changes:
- https://github.com/renovatebot/renovate/blob/b376b8e0c0790b7d349c08e068eef016e03511d5/lib/modules/manager/regex/readme.md?plain=1#L183-L184
- https://github.com/renovatebot/renovate/blob/b376b8e0c0790b7d349c08e068eef016e03511d5/docs/usage/configuration-options.md?plain=1#L1731
- https://github.com/renovatebot/renovate/blob/b376b8e0c0790b7d349c08e068eef016e03511d5/docs/usage/configuration-options.md?plain=1#L2700
-

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**